rat/codegen_backends.py

```diff
@@ -1,13 +1,18 @@
+import jax
```

Comment:
This file is becoming a bit of a catch-all. I kinda want to break it up.

Reply:
Outside of some special high-order functions(`vmap, jit`), aren't most of them in `jax.numpy`?

Reply:
Yah, but I meant more about the contents of codegen_backends. This file is becoming a dumping ground for walkers that are used in more than one place and the file is getting kinda long. The relationships between things within this file isn't clear.

Reply:
Ah yes agreed